gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -409,6 +409,10 @@ class {{service.name}}RestTransport({{service.name}}Transport):
 
     {% endfor %}
 
+    @property
+    def kind(self) -> str:
```

Comment:
Could you use isinstance to determine this ?

Reply:
We're trying to do something here that I don't think `isinstance` is a good choice for.
1) After construction, the client doesn't know anything about the type of the transport.
a) There are situations where the client doesn't know the type of the transport _during_ construction: if the transport is injected, or if no transport is given and the default type needs to be used.
2) The first time an extended LRO method is called, the client needs to instantiate an extended operations client, and it **SHOULD** use a corresponding transport type: grpc if the normal client's transport is grpc, rest if the normal client's transport is rest.
a) Because of 1a, we can't solve this problem by making the extended operations client eagerly or by storing the transport "kind" during construction.
3) We therefore need a mechanism to determine the "kind" of transport the normal client has, i.e. grpc or rest. If we use `isinstance`, we still need a way to turn that information into a decision about what kind of transport to make or the extended operations client. Something like
```python
ex_op_transport = None
{% if "rest" in opts.transports %}
if isinstance(self._transport, {{ service.name }}RestTransport):
    ex_op_transport = "rest"
{% endif %}
{% if "grpc" in opts.transports %}
if isinstance(self._transport, {{ service.name }}GrpcTransport):
    ex_op_transport = "grpc"
{% endif %}
if not ex_op_transport:
    raise ValueError(f"Unexpected transport type: {type(self._transport)}")

self._extended_operations_client = {{ api.get_extended_operation_service(m).name }}(transport=ex_op_transport)
```
That looks a little clunky, IMO, compared to the logic where the transports self identify their "kind" and route that through.
```python
# Something like this, it's elsewhere in the diff
self._extended_operations_client = {{ api.get_extended_operation_service(m).name }}(transport=self._transport.kind)
```
4) If we ever need to add another transport "kind", having this `kind` property lets us handle that while making fewer code changes. Static analysis will detect whether the new transport class defines an override for `kind`, and none of the rest of the logic needs to be modified. If we had the `isinstance` code block above, new transport classes would need to be explicitly added to the logic. Heck, it should be possible for the user to define transport classes for both the normal client and the extended operations client and use them together without any code changes in the client class.